server/index.js

```diff
@@ -133,6 +133,23 @@ app.patch('/api/triptodo/:todoId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/itemscompleted/:tripId', (req, res, next) => {
+  const tripId = parseInt(req.params.tripId, 10);
+  if (!tripId) {
+    throw new ClientError(400, 'tripId must be a positive integer');
+  }
+  const sql = `
+    select count("completed") as "itemsCompleted"
+      from "todo"
+     where "tripId" = $1
+       and "completed" = 'true'
+  `;
+  const params = [tripId];
+  db.query(sql, params)
+    .then(result => res.json(result.rows[0].itemsCompleted))
+    .catch(err => next(err));
+});
```

Comment:
Try retrieving this data at the same time as when you retrieve your trip using a `join`.